DEVELOPER.md

```diff
@@ -128,15 +137,34 @@ To run single unit test
 TF_ACC=0 go test -v ./... -run TestUnit<test_name>
 ```
 
-Note: The tests require permissions on the folders, these permissions are assigned when creating your container.
-If you have permission problems when running the unit tests, you can rebuild your development container
-or run the following commands again to assign the permissions to the necessary folders.
+> [!NOTE]
+> The tests require permissions on the folders, these permissions are assigned when creating your container. If you have permission problems when running the unit tests, you can rebuild your development container or run the following commands again to assign the permissions to the necessary folders.
 
 ```bash
 sudo chown -R vscode /workspaces/terraform-provider-power-platform/
 sudo chown -R vscode /go/pkg
 ```
 
+## Writing Tests
+
+All the test for a given resource/datasource are located in `/internal/powerplatform/<resource/datasource_name>_test.go` file. When writing a new feature you should try to create [happy path](https://en.wikipedia.org/wiki/Happy_path) test(s) for you feature covering create, read and deletion of your new feature. For updates you should cover not only update of all properties but situation when a force recreate of a resource is requried (if you have such propeties in you resource).
+
+### Writing Unit Tests
+
+Unit test are created by mocking HTTP request, some of the often used HTTP mocks encapsulated in `ActivateEnvironmentHttpMocks` function, so that you don't have to write them for every test.
+
+when implementing new mocks, the mokcked response json files should be located in `/internal/powerplatform/services/<your_service_name>/test/<resource_or_datasource>/<name_of_the_unit_test>` folder
```

Comment:
```suggestion
Unit test are created by mocking HTTP request, some of the often used HTTP mocks encapsulated in `ActivateEnvironmentHttpMocks` function, so that you don't have to write them for every test. When implementing new mocks, the mokcked response json files should be located in `/internal/powerplatform/services/<your_service_name>/test/<resource_or_datasource>/<name_of_the_unit_test>` folder
```